idea/src/org/jetbrains/kotlin/idea/intentions/JoinDeclarationAndAssignmentIntention.kt

```diff
@@ -140,6 +142,10 @@ class JoinDeclarationAndAssignmentIntention : SelfTargetingRangeIntention<KtProp
 
         val firstAssignment = assignments.firstOrNull() ?: return null
         if (assignments.any { it !== firstAssignment && it.parent.parent is KtSecondaryConstructor }) return null
+        if (!property.isLocal &&
+            firstAssignment.parent != propertyContainer &&
+            firstAssignment.right?.anyDescendantOfType<KtNameReferenceExpression> { it.mainReference.resolve() is KtParameter } == true
```

Comment:
Please reuse `context` to avoid additional resolve via `mainReference.resolve()`.